File: abstractions/ruby/microsoft_kiota_abstractions/lib/microsoft_kiota_abstractions/api_client_builder.rb
require_relative "serialization/parse_node_factory"
require_relative "serialization/parse_node_factory_registry"
require_relative "serialization/serialization_writer_factory"
require_relative "serialization/serialization_writer_factory_registry"

module MicrosoftKiotaAbstractions
  class ApiClientBuilder
    def self.register_default_serializer(factory_class)
      factory = factory_class.new()
      MicrosoftKiotaAbstractions::SerializationWriterFactoryRegistry.default_instance.content_type_associated_factories[factory.get_valid_content_type()] = factory
    end

    def self.register_default_deserializer(factory_class)
      factory = factory_class.new()
      MicrosoftKiotaAbstractions::ParseNodeFactoryRegistry.default_instance.content_type_associated_factories[factory.get_valid_content_type()] = factory
    end
  end
end
  

File: abstractions/swift/Source/MicrosoftKiotaAbstractions/ApiClientBuilder.swift
public class ApiClientBuilder {
    private init() {

    }
    private static let defaultSerializationWriterFactoryInstanceIntl = SerializationWriterFactoryRegistry()
    public static var defaultSerializationWriterFactoryInstance: SerializationWriterFactory {
        get {
            return defaultSerializationWriterFactoryInstanceIntl
        }
    }
    public static func registerDefaultSerializer(metaFactory: () -> SerializationWriterFactory) {
        let factory = metaFactory()
        if let contentType = try? factory.getValidContentType() {
            if contentType != "" {
                defaultSerializationWriterFactoryInstanceIntl.contentTypeAssociatedFactories[contentType] = factory
            }
        }
    }
    private static let defaultParseNodeFactoryInstanceIntl = ParseNodeFactoryRegistry()
    public static var defaultParseNodeFactoryInstance: ParseNodeFactory {
        get {
            return defaultParseNodeFactoryInstanceIntl
        }
    }
    public static func registerDefaultParser(metaFactory: () -> ParseNodeFactory) {
        let factory = metaFactory()
        if let contentType = try? factory.getValidContentType() {
            if contentType != "" {
                defaultParseNodeFactoryInstanceIntl.contentTypeAssociatedFactories[contentType] = factory
            }
        }
    }
}